common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsFileHelper.kt

```diff
@@ -0,0 +1,42 @@
+package com.revenuecat.purchases.common.diagnostics
+
+import com.revenuecat.purchases.common.FileHelper
+import com.revenuecat.purchases.common.verboseLog
+import org.json.JSONObject
+
+/**
+ * All methods in this file should be executed within the diagnostics thread to ensure there are no threading issues.
+ */
+class DiagnosticsFileHelper(
+    private val fileHelper: FileHelper
+) {
+    companion object {
+        const val DIAGNOSTICS_FILE_PATH = "RevenueCat/diagnostics/diagnostic_events.jsonl"
+    }
+
+    @Synchronized
+    fun appendEventToDiagnosticsFile(diagnosticsEvent: DiagnosticsEvent) {
+        fileHelper.appendToFile(DIAGNOSTICS_FILE_PATH, diagnosticsEvent.toString() + "\n")
+    }
+
+    @Synchronized
+    fun cleanSentDiagnostics(diagnosticsSentCount: Int) {
+        fileHelper.removeFirstLinesFromFile(DIAGNOSTICS_FILE_PATH, diagnosticsSentCount)
+    }
+
+    @Synchronized
+    fun deleteDiagnosticsFile() {
+        if (!fileHelper.deleteFile(DIAGNOSTICS_FILE_PATH)) {
+            verboseLog("Failed to delete diagnostics file.")
+        }
+    }
+
+    @Synchronized
```

Comment:
Note that all these need to be synchronized. The idea is that all this code executes in a single background thread with a lower priority than other threads. To avoid race conditions, all these should run in that same thread.